spi/src/transformations/user.ts

```diff
@@ -29,39 +29,54 @@ export const transformUsers = async (connection: IDBConnection): Promise<void> =
     -------------------------------------------------------------------------------------------------
     -- Turn deduplicated users into SIMS users
     -------------------------------------------------------------------------------------------------
+    WITH w_existing_users AS (
+        SELECT 
+            system_user_id,
+            family_name,
+            given_name
+        FROM biohub.system_user
+    )
     INSERT INTO biohub.system_user (
-      user_identity_source_id,
-      user_identifier,
-      record_effective_date,
-      display_name,
-      given_name,
-      family_name,
-      notes,
-      email
+        user_identity_source_id,
+        user_identifier,
+        record_effective_date,
+        display_name,
+        given_name,
+        family_name,
+        notes,
+        email
     )
     SELECT 
-      (SELECT user_identity_source_id FROM biohub.user_identity_source WHERE name = 'UNVERIFIED'),
-      'spi-' || id,
-      now(),
-      display_name,
-      given_name,
-      family_name,
-      'Migrated from SPI as user' || id,
-      'default'
+        (SELECT user_identity_source_id FROM biohub.user_identity_source WHERE name = 'UNVERIFIED'),
+        'spi-' || md.id,
+        now(),
+        md.display_name,
+        md.given_name,
+        md.family_name,
+        'Migrated from SPI as user ' || md.id,
+        COALESCE(spp.email_address, 'default') AS email
     FROM 
-      migrate_spi_user_deduplication;
+        migrate_spi_user_deduplication md
+    LEFT JOIN 
+        public.spi_secure_persons spp ON spp.first_name = md.given_name AND spp.last_name = md.family_name
+    LEFT JOIN 
+        w_existing_users eu ON eu.given_name = md.given_name AND eu.family_name = md.family_name
+    WHERE eu.system_user_id IS NULL;
```

Comment:
eu.system_user_id will never be null because it is the primary key

Reply:
We are left-joining eu onto deduplication table (which is the from table in the insert statement) -- in this case, everything from deduplication is joined to only matching records in eu, wouldnt therefore any deduplication records without a match in eu have a null eu.system_user_id as they haven't been "inserted" yet?